flight/Libraries/math/misc_math.c

```diff
@@ -192,8 +192,8 @@ void vector2_clip(float *vels, float limit)
 	float scale = mag / limit;
 
 	if (scale > 1) {
-		vels[0] /= mag;
-		vels[1] /= mag;
+		vels[0] /= scale;
+		vels[1] /= scale;
```

Comment:
Maybe.  Fast-math lets the compiler do the inverse calculation.  In the worst case it is only a few cycles on something with a fpu, and maybe not even that if loads are executed at the same time or the compiler is feeling smart.

Moving a lot of this stuff as inline into the header should be a big win though--- because the compiler is free to reorder code over larger boundaries and eliminate larger common subexpressions.


Sent from my iPhone

> On Sep 18, 2015, at 6:21 AM, Kenn Sebesta <notifications@github.com> wrote:
> 
> In flight/Libraries/math/misc_math.c:
> 
> > @@ -192,8 +192,8 @@ void vector2_clip(float *vels, float limit)
> >  	float scale = mag / limit;
> >  
> >  	if (scale > 1) {
> > -		vels[0] /= mag;
> > -		vels[1] /= mag;
> > +		vels[0] /= scale;
> > +		vels[1] /= scale;
> Wouldn't this whole function be better-- i.e. more efficient-- as
> 
> float mag = vectorn_magnitude(vels, 2);    // only horiz component
> 
> if (mag > limit) {
>     vels[0] *= (limit / mag);
>     vels[1] *= (limit / mag);
> }
> —
> Reply to this email directly or view it on GitHub.
>